src/plugins/privacy/tjvendors/tjvendors.php

```diff
@@ -0,0 +1,276 @@
+	 */
+	private function createVendorFessDomain(JTableUser $user)
+	{
+		$domain = $this->createDomain('TJvendor user fees', 'TJvendor user fees data');
+
+		$query = $this->db->getQuery(true)
+			->select('vendor_id')
+			->from($this->db->quoteName('#__tjvendors_vendors'))
+			->where($this->db->quoteName('user_id') . ' = ' . (int) $user->id);
+
+		$vendorId = $this->db->setQuery($query)->loadResult();
+
+		if ($vendorId)
+		{
+			$query1 = $this->db->getQuery(true)
+					->select('id, vendor_id, currency, client, percent_commission, flat_commission')
+					->from($this->db->quoteName('#__tjvendors_fee'))
+					->where($this->db->quoteName('vendor_id') . ' = ' . (int) $vendorId);
+			$userFeesData = $this->db->setQuery($query1)->loadAssocList();
+
+			if (!empty($userFeesData))
+			{
+				foreach ($userFeesData as $feeData)
+				{
+					$domain->addItem($this->createItemFromArray($feeData, $feeData['id']));
+				}
+			}
+		}
+
+		return $domain;
+	}
+
+	/**
+	 * Create the domain for the TJvendor to get vendor passbook data
+	 *
+	 * @param   JTableUser  $user  The JTableUser object to process
+	 *
+	 * @return  PrivacyExportDomain
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	private function createVendorPassbookDomain(JTableUser $user)
+	{
+		$domain = $this->createDomain('TJvendor user passbook', 'TJvendor user passbook data');
+
+		$query = $this->db->getQuery(true)
+			->select('vendor_id')
+			->from($this->db->quoteName('#__tjvendors_vendors'))
+			->where($this->db->quoteName('user_id') . ' = ' . (int) $user->id);
+
+		$vendorId = $this->db->setQuery($query)->loadResult();
+
+		if ($vendorId)
+		{
+			$query1 = $this->db->getQuery(true)
+					->select('id, vendor_id, currency, total, credit, debit, reference_order_id, transaction_time, client, transaction_id, status, params')
+					->from($this->db->quoteName('#__tjvendors_passbook'))
+					->where($this->db->quoteName('vendor_id') . ' = ' . (int) $vendorId);
+			$userPassbookData = $this->db->setQuery($query1)->loadAssocList();
+
+			if (!empty($userPassbookData))
+			{
+				foreach ($userPassbookData as $passbookData)
+				{
+					$domain->addItem($this->createItemFromArray($passbookData, $passbookData['id']));
+				}
+			}
+		}
+
+		return $domain;
+	}
+
+	/**
+	 * Performs validation to determine if the data associated with a remove information request can be processed
+	 *
+	 * This event will not allow a super user account to be removed
+	 *
+	 * @param   PrivacyTableRequest  $request  The request record being processed
+	 * @param   JUser                $user     The user account associated with this request if available
+	 *
+	 * @return  PrivacyRemovalStatus
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onPrivacyCanRemoveData(PrivacyTableRequest $request, JUser $user = null)
+	{
+		$status = new PrivacyRemovalStatus;
+
+		$status->canRemove = false;
+		$status->reason    = JText::_('PLG_PRIVACY_TJVENDORS_ERROR_CANNOT_REMOVE_USER_DATA');
```

Comment:
If vendor exists then only return false, in this case this will always return false. @deepa-g